src/detect-transform-compress-whitespace.c

```diff
@@ -111,7 +111,11 @@ static void TransformCompressWhitespace(InspectionBuffer *buffer, void *options)
         return;
     }
 
-    uint8_t output[input_len]; // we can only shrink
+    // we can only shrink
+    uint8_t *output = (uint8_t *)InspectionBufferCheckAndExpand(buffer, input_len);
```

Comment:
why do we need the cast, instead of updating `InspectionBufferCheckAndExpand` to return `uint8_t *`?

Reply:
nice